builder/builder.go

```diff
@@ -333,8 +324,17 @@ func (h *HeighlinerBuilder) buildChainNodeDockerImage(
 		} else {
 			baseVersion = baseVer
 		}
+
+		if h.race {
+			buildEnv += " GOFLAGS=-race"
```

Comment:
This should work on a plain `go build` command, but if we are performing the build through a third party Makefile, there is a chance that their file could hardcode a value for `GOFLAGS`, overriding what we've provided.

It would be good if there was a way to run `go version -m OUTPUT_BINARY` and check for the presence of `build	-race=true` -- ideally failing this command if the resulting binary doesn't match.

Reply:
Most cosmos chains build with a Makefile `make install`. The `build-target` in `chains.yaml` can be a multi-line build script, so I think it would be best to avoid assumptions there. We could still have problems if a Makefile is overriding `GOFLAGS` and not including the existing var.

Checking the output binary(ies) can be added to the Dockerfile. Since non-go binaries might also be included, we could ignore `not a Go executable` errors from `go version`, which will ensure that all resulting go bins have the race detector enabled.

Reply:
Assertions have been added for the race detector in the output go bins when `--race` is set.